fpmsyncd/routesync.cpp

```diff
@@ -733,6 +733,32 @@ void RouteSync::onRouteMsg(int nlmsg_type, struct nl_object *obj, char *vrf)
         {
             SWSS_LOG_DEBUG("Skip routes to eth0 or docker0: %s %s %s",
                     destipprefix, gw_list.c_str(), intf_list.c_str());
+            // If intf_list has only this interface, that means all of the next hops of this route 
+            // have been removed and the next hop on the eth0/docker0 has become the only next hop. 
+            // In this case since we do not want the route with next hop on eth0/docker0, we return. 
+            // But still we need to clear the route from the APPL_DB. Otherwise the APPL_DB and data 
+            // path will be left with stale route entry
+            if(alsv.size() == 1)
```

Comment:
From the context, it seems for del msg, should we check it by
if((nlmsg_type == RTM_DELROUTE) && (alsv.size() == 1))
?

Reply:
Yes.

Reply:
@vganesan-nokia , do you want to address the comment?

Reply:
Even it is a RTM_NEWROUTE on eth0/docker0, it may run into

if (!warmRestartInProgress) 
{
  ... 
  m_routeTable.del(destipprefix)
}

then the route is deleted from APPL_DB and ASIC_DB ?
It happened on my real device with  default route (0.0.0.0/0) is removed.

Reply:
> @vganesan-nokia , do you want to address the comment?

Yes. I'll make a PR to fix this.